course.py

```diff
@@ -0,0 +1,262 @@
+from db import db, User, Course, Teacher, Organization, Configuration
+from flask import Blueprint, render_template, flash, url_for, request, make_response, redirect, \
+    Flask, jsonify, session
+from util import get_current_year
+import json, re
+
+course_bp = Blueprint('course', __name__)
+
+
+def validate_course_code(code):
+    course_code_regex = r'^[a-zA-Z0-9\s]+$'
+    return re.match(course_code_regex, code) is not None
+
+
+def validate_course_title(title):
+    title_regex = r'^[\w\s,:\']+$'
+    return re.match(title_regex, title) is not None
+
+
+def validate_string_pattern(string):
+    pattern = r'^[a-zA-Z\s]+$'
+    return re.match(pattern, string) is not None
+
+
+def validate_number_pattern(number):
+    pattern = r'^(0|[1-9][0-9]*)$'
+    return re.match(pattern, number) is not None
+
+
+@course_bp.route('/form_course')
+@login_required
+def form_course():
+    return render_template('add_course.html')
+
+def validate_form_data(form, extra_fields_needed=False):
+    mandatory_fields = {
+        'code': validate_course_code,
+        'title': validate_course_title,
+        'language': validate_string_pattern,
+        'year': validate_number_pattern,
+        'quadri': validate_number_pattern,
+    }
+
+    extra_fields = {
+        'nbr_students': validate_number_pattern,
+        'nbr_teaching_assistants': validate_number_pattern,
+        'nbr_monitor_students': validate_number_pattern,
+    }
+
+    for field, validator in mandatory_fields.items():
+        value = form.get(field)
+        if not value:
+            return f"Missing {field}", 400
+        if not validator(value):
+            return f"Invalid {field}", 400
+
+    if extra_fields_needed:
+        for field, validator in extra_fields.items():
+            value = form.get(field)
+            if not value:
+                return f"Missing {field}", 400
+            if not validator(value):
+                return f"Invalid {field}", 400
+
+def assign_teachers_to_course(course_id, course_year, assigned_teachers):
+    try:
+        for teacher_id in assigned_teachers:
+            existing_teacher = db.session.query(Teacher).filter(Teacher.course_id == course_id,
+                                                                Teacher.user_id == teacher_id,
+                                                                Teacher.course_year == course_year).first()
+
+            if existing_teacher is None:
+                new_teacher = Teacher(user_id=teacher_id, course_id=course_id, course_year=course_year)
+                db.session.add(new_teacher)
+        db.session.commit()
+    except Exception as e:
+        db.session.rollback()
+        raise e
+
+
+@course_bp.route('/add_course', methods=['POST'])
+@login_required
+def add_course():
+    form = request.form
+    if not form:
+        return make_response("Problem with form request", 500)
+
+    error = validate_form_data(form)
+    if error:
+        return make_response(error[0], error[1])
```

Comment:
```suggestion
        return make_response(*error)
```
The `*` operator explodes the tuple and allows to use the tuple values as positional arguments.
The `**` operator does the same thing for dictionaries and named arguments.